calour/tests/test_export_html.py

```diff
@@ -0,0 +1,42 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--,  Calour development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from unittest import main
+from tempfile import mkdtemp
+from os.path import join
+import shutil
+
+from calour._testing import Tests
+import calour as ca
+
+
+class ExportHtmlTests(Tests):
+    def setUp(self):
+        super().setUp()
+        self.test1 = ca.read(self.test1_biom, self.test1_samp, normalize=None)
+
+    def test_export_html(self):
+        exp = self.test1
+        d = mkdtemp()
+        f = join(d, 'calour_interactive_heatmap.html')
+        exp.export_html(sample_field='group', output_file=f, feature_field='taxonomy', title='the heatmap')
+        # load the results
+        with open(f) as output_fl:
+            output_data = output_fl.readlines()
+        with open(join(self.test_data_dir, 'export_html_result.html')) as expected_fl:
+            expected_data = expected_fl.readlines()
+        # we cannot test equality of the whole files since plot output is different
+        # on different machines
```

Comment:
why it is not deterministic? any randomization in the process?

Reply:
It's deterministic, but we store the heatmap as png and embed it into the html. and i think the png is different between different os/computer configs (i.e. bit depth, exact colors, etc.)
when i compared the complete html, it works on my local machine but not on travis, and the problem is in the embedded image...
so no we just test the first 100 lines which contain sample ids etc.